### PR TITLE
Add Central Bank of The Gambia provider

### LINKED ISSUE #415: Add Central Bank of The Gambia (CBG) provider
## Central Bank of The Gambia (CBG)

**Base currency:** GMD

### API

- **Format:** JSON
- **Auth:** None. Plain nginx/PHP. No CDN/WAF/geo-block.
- **Endpoints:**
  - Per-currency time series: GET https://www.cbg.gm/ajax/indicative-exchange-rates/{ISO}
  - Latest table: https://www.cbg.gm/indicative-exchange-rates-latest
- Response: JSON array of [epoch_ms, rate] pairs.

### Coverage

- 33 quote currencies: USD, EUR, GBP, CHF, SEK, CAD, XOF, NOK, DKK, SAR, JPY, AUD, TWD, LKR, THB, PHP, NZD, AED, KWD, NGN, HKD, ZAR, EGP, CNY, BRL, INR, GHS, SLL, TRY, GNF, WAUA, XDR, SGD
- USD/EUR/GBP from 2000-01-07; XOF from 2005; XDR/SLL/GNF from 2019-11
- Cadence: weekly (~52/yr) 2000-2023, business-daily from 2024-01 onward
- Latest update same-day

### Direction

Source records "1 FOREIGN = X GMD" (e.g. 72.39 GMD per USD). Base = foreign ISO, quote = GMD. Adapter inverts per Frankfurter convention.

### License

Footer only: "Copyright (c) 2026. All Rights Reserved." No explicit redistribution permission, no commercial/no-derivative prohibition. Same implicit-public-statistic posture as several providers we ship.

### Caveats

- 2000-2023 was weekly cadence. From 2024 onward business-daily. Faithful relay (per project_data_philosophy); don't carry-forward during the weekly era for V2 range queries.
- WAUA (West African Unit of Account) and XDR are composite units; exclude.
- Branded "Daily Valuation Rates" since 2024 transition.

### Notes

- Regional African basket (XOF, NGN, GHS, SLL, GNF) is a useful add.
- Closes WAMZ-zone coverage gap alongside BSL Sierra Leone (#391).

### Checklist

See implementing-providers skill.

### PR DESCRIPTION
**Pull request by lineoffligbot**
## Summary

Adds CBG as a data provider. Indicative daily valuation rates for 32 currencies against the Gambian dalasi (GMD), with USD/EUR/GBP archives back to 2000-01-07, XOF from 2005, and the remaining currencies from 2019-11-18.

## Source

- Per-currency JSON time series: `https://www.cbg.gm/ajax/indicative-exchange-rates/{ISO}`
- Each endpoint returns the full archive as `[[epoch_ms, rate], ...]`. The adapter iterates the known currency list and filters by date in memory.
- No auth, plain nginx/PHP backend.

## Coverage and cadence

- `coverage_start`: 2000-01-07
- `publish_cadence`: `daily`. Cadence was weekly (about 52 entries per year) from 2000 through 2023, then business-daily from 2024-01 onward. We relay the published series as-is; weekly-era gaps are preserved.
- `publish_schedule`: `*/30 11-13 * * 1-5`. Best estimate of the daily publish window. The page lists no fixed publication time, so the 3-hour polling window starts at 11:00 UTC.

## Currencies

32 ISO 4217 codes: USD, EUR, GBP, CHF, SEK, CAD, XOF, NOK, DKK, SAR, JPY, AUD, TWD, LKR, THB, PHP, NZD, AED, KWD, NGN, HKD, ZAR, EGP, CNY, BRL, INR, GHS, SLL, TRY, GNF, XDR, SGD.

## Quirks

- Provider records `1 FOREIGN = X GMD`, so foreign currency is `base` and GMD is `quote` (same convention as NBG, BBK, CBN).
- WAUA (West African Unit of Account) is published but not ISO 4217; the adapter does not request it.
- The AJAX endpoint returns rates at two-decimal precision. Very-high-denomination currencies (GNF) floor to 0.01. We store what CBG publishes via this endpoint.
- SLL series shows the 2022 redenomination as a real currency event: pre-2022 values of 0.01 GMD (old leone), zeros during the changeover window (dropped on ingest by the Provider model's non-positive filter), then 4.x values from July 2022 onward (new leone).

## Sanity check vs Wise (2026-05-22)

| Pair | CBG (GMD) | Wise (GMD) | Deviation |
|------|-----------|------------|-----------|
| USD  | 72.39     | 73.24      | -1.16%    |
| EUR  | 86.28     | 84.93      | +1.58%    |
| GBP  | 97.99     | 98.29      | -0.31%    |

Within institutional vs real-time spread for an indicative central-bank rate.

Closes #415

## Test plan

- [x] `APP_ENV=test bundle exec rake` (lint + tests)
- [x] VCR-backed adapter spec covers fetch, date filtering, parse, and edge cases
- [x] Live fetch verified against the upstream JSON endpoint
- [x] Rate direction confirmed against Wise mid-market for USD/EUR/GBP

### CODE REVIEW

File: lib/provider/adapters/cbg.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        data.filter_map do |entry|
+          next unless entry.is_a?(Array) && entry.size == 2
+
+          epoch_ms, value = entry
+          next if epoch_ms.nil? || value.nil?
+
+          rate = Float(value)
+          next if rate.zero?
+
+          date = Time.at(Integer(epoch_ms) / 1000).utc.to_date
```
[the remark this answers is not in the document]
Comment (lineoffligbot) [reply lib/provider/adapters/cbg.rb:93]: Convention here is to let parse failures bubble up rather than rescue them. The scheduler retries, and surfacing the error is how we notice when an upstream changes shape. Rescuing would hide that signal, so leaving as is.
